File: inventario-backend/src/main/java/cl/uddi/inventario/domain/Movimiento.java
package cl.uddi.inventario.domain;

import java.time.Instant;

public abstract class Movimiento {
    public Instant fecha = Instant.now();
    public MovementType tipo;
    public int sku;
    public int cantidad;     
    public int precioUnit;
    public String doc;
    public String nota;
    public String rutUsuario;

    protected Movimiento(MovementType tipo, int sku, int cantidad, int precioUnit,
                         String doc, String nota, String rutUsuario) {
        this.tipo = tipo; this.sku = sku; this.cantidad = cantidad; this.precioUnit = precioUnit;
        this.doc = doc; this.nota = nota; this.rutUsuario = rutUsuario;
    }

    public abstract int aplicar(int stockActual);
}

File: Bootcamp Frontend/Models/Movimiento.cs
using System;
using System.Text.Json.Serialization;

namespace Bootcamp_Frontend.Models
{
    public class Movimiento
    {
        [JsonPropertyName("sku")]
        public int Sku { get; set; }

        [JsonPropertyName("cantidad")]
        public int Cantidad { get; set; }

        [JsonPropertyName("precioUnitario")]
        public int PrecioUnitario { get; set; }

        [JsonPropertyName("rutUsuario")]
        public string RutUsuario { get; set; }

        [JsonPropertyName("documento")]
        public string Documento { get; set; }

        [JsonPropertyName("nota")]
        public string Nota { get; set; }

        [JsonPropertyName("tipo")]
        public string Tipo { get; set; }

        [JsonPropertyName("fecha")]
        public string Fecha { get; set; }

        public int Total => Cantidad * PrecioUnitario;
    }
}
